web/client/plugins/TOC/components/Legend.jsx

```diff
@@ -114,7 +118,23 @@ class Legend extends React.Component {
     }
     render() {
         if (!this.state.error && this.props.layer && this.props.layer.type === "wms" && this.props.layer.url) {
-            return <img onError={this.onImgError} onLoad={(e) => this.validateImg(e.target)} src={this.getUrl(this.props)} style={this.props.style}/>;
+            const url = this.getUrl(this.props);
+            const authMethod = getAuthenticationMethod(url);
+            return (
+                authMethod === "bearer" ?
+                    <SecureImage
+                        onError={this.onImgError}
+                        onLoad={(e) => this.validateImg(e.target)}
+                        src={url}
+                        style={this.props.style}
+                    /> :
+                    <img
+                        onError={this.onImgError}
+                        onLoad={(e) => this.validateImg(e.target)}
+                        src={url}
+                        style={this.props.style}
+                    />
+            );
```

Comment:
Instead to have a if method could we move the logic inside SecureImage directly, in this way the component can decide if add it as query parameter or as bearer header

Reply:
sounds good